tmail-backend/jmap/extensions-api/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -92,22 +109,32 @@ object MinAutoCompleteInputLength {
 case class MinAutoCompleteInputLength(value: Int)
 
 trait EmailAddressContactSearchEngine {
-  def index(accountId: AccountId, fields: ContactFields): Publisher[EmailAddressContact]
+  def index(accountId: AccountId, fields: ContactFields): Publisher[EmailAddressContact] =
+    index(accountId, fields, None)
+
+  def index(accountId: AccountId, fields: ContactFields, addressBookCardUid: Option[String]): Publisher[EmailAddressContact]
 
   def index(domain: Domain, fields: ContactFields): Publisher[EmailAddressContact]
 
-  def update(accountId: AccountId, updatedFields: ContactFields): Publisher[EmailAddressContact]
+  def update(accountId: AccountId, updatedFields: ContactFields): Publisher[EmailAddressContact] =
+    update(accountId, updatedFields, None)
+
+  def update(accountId: AccountId, updatedFields: ContactFields, addressBookCardUid: Option[String]): Publisher[EmailAddressContact]
 
   def update(domain: Domain, updatedFields: ContactFields): Publisher[EmailAddressContact]
 
   def delete(accountId: AccountId, mailAddress: MailAddress): Publisher[Void]
 
+  def delete(accountId: AccountId, mailAddress: MailAddress, addressBookCardUid: String): Publisher[Void]
+
   def delete(domain: Domain, mailAddress: MailAddress): Publisher[Void]
 
   def autoComplete(accountId: AccountId, part: String, limit: Int = 256): Publisher[EmailAddressContact]
 
   def list(accountId: AccountId): Publisher[EmailAddressContact]
 
+  def list(accountId: AccountId, addressBookCardUid: String): Publisher[EmailAddressContact]
```

Comment:
This method is necessary for the Sabre contact consumer to identify `toAdd`, `toUpdate`, and `toDelete` contacts from "AMQP message" on the `sabre:contact:updated` exchange.